File: Z-fichier_test_compilation/ram_consommation/ram.java
import java.util.ArrayList;

public class ram {
    public static void main(String[] args) {
        ArrayList<byte[]> memory = new ArrayList<>();
        try {
            for (int i = 0; i < 600; i++) {
                memory.add(new byte[1024 * 1024]); // Allocate 1MB
            }
            System.out.println("Consumed 600MB of memory");
            Thread.sleep(60000); // Sleep for 1 minute
        } catch (InterruptedException e) {
            System.err.println("Program interrupted");
        }
    }
}

File: Z-fichier_test_compilation/ram_consommation/ram.py
import time

def allocate_memory():
    memory = [bytearray(1024 * 1024) for _ in range(600)]  # Allocate 600MB
    print("Consumed 600MB of memory")
    time.sleep(60)  # Sleep for 1 minute

if __name__ == "__main__":
    allocate_memory()
